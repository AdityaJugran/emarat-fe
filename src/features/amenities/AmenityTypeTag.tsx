import { Tag } from 'antd';
import { AmenityType } from './Types';

type AmenityTypeTagType = {
	type: AmenityType['type'];
	fix?: boolean;
	className?: string;
};

const amenityTypeColorMapper = {
	basic: 'green',
	flexible: 'gold',
	limited: 'red',
};

export default function AmenityTypeTag({
	type,
	fix,
	className,
}: AmenityTypeTagType) {
	return (
		<Tag
			color={amenityTypeColorMapper[type]}
			className={`font-semibold capitalize rounded mr-0 ${className}`}
		>
			{type}
		</Tag>
	);
}
