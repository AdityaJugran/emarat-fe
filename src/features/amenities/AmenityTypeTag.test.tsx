import AmenityTypeTag from 'features/amenities/AmenityTypeTag';
import renderer from 'react-test-renderer';

it('Amenity Type Tag', () => {
	const tree = renderer
		.create(<AmenityTypeTag type="basic" className="h-fit" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
