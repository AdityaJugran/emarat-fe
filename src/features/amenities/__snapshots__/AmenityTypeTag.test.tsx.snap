// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amenity Type Tag 1`] = `
<span
  className="ant-tag ant-tag-green font-semibold capitalize rounded mr-0 h-fit"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  basic
</span>
`;
